docs/src/test-assertions-js.md

```diff
@@ -276,12 +276,12 @@ export const expect = baseExpect.extend({
     }
 
     const message = pass
-      ? () => this.utils.matcherHint(assertionName, undefined, undefined, { isNot: this.isNot }) +
+      ? () => this.utils.matcherHint(assertionName, undefined, undefined, { isNot: true }) +
```

Comment:
How is thinking about `pass == isNot` better than just passing `isNot` for `isNot`?

Reply:
The behaviour around when `message` is respected confused some users. They thought that if a matcher passes, then `message` is respected. That's what this clarification is about. Passing static `isNot` makes this behaviour more explicit. Maybe we should also add some docs copy here.

Reply:
I've updated the copy in https://github.com/microsoft/playwright/pull/33321/commits/946338d572906e933474d6a64d5eb24a62470ac8, let me know what you think

Reply:
I still prefer a direct `isNot: this.isNot` here. Do not encourage users to do a mental math and compute the value here, it is important that isNot is this.isNot and nothing else. Passing it as is is simpler than pre-computing it for the computer, less error-prone. In our own matchers we use a following pattern:

```js
const matcherOptions = {
  isNot: this.isNot,
  promise: this.promise,
};
```
followed by

```js
matcherHint(this, receiver, matcherName, receiver, expected, matcherOptions)
```

That way we never think about it, no matter which branch of message is computed.

Reply:
Alright, i've reverted it. Do you feel the same about the string tempalting in line 282?